.github/scripts/utils/utils.ts

```diff
@@ -0,0 +1,120 @@
+import { execSync } from 'child_process';
+import fs from 'fs';
+
+const owner = process.env.OWNER;
+const repository = process.env.REPOSITORY;
+
+/**
+ * Downloads an artifact from CircleCI.
+ * @param branch - The branch name.
+ * @param headCommitHash - The commit hash of the branch.
+ * @param artifactName - The name of the artifact to download.
+ * @param outputFilePath - The path to save the downloaded artifact.
+ * @param jobName - The name of the job that produced the artifact.
+ */
+export async function downloadCircleCiArtifact(branch: string, headCommitHash: string, artifactName: string, outputFilePath: string, jobName: string): Promise<void> {
+    console.log('Branch', branch);
+    console.log('Commit', headCommitHash);
+    console.log('Owner', owner);
+    console.log('Repository', repository);
+    console.log('url', `https://circleci.com/api/v2/project/gh/${owner}/${repository}/pipeline?branch=${branch}`);
+
```

Comment:
leaving the console logs intentionally to make debugging easier in case it's needed